ui/src/components/SampleView/DrawGridPlugin.js

```diff
@@ -401,7 +407,10 @@ export default class DrawGridPlugin {
       }
 
       if (!this.drawing) {
-        if (this.gridResultFormat === 'RGB') {
+        if (
+          this.gridResultFormat === 'PNG' ||
```

Comment:
No sure exactly what you would like to achieve here, how are you using the PNG results ?. This logic is only used to display RGB results, creating a "cell" with the corresponding RGB value for each value in the result array. The PNG image is simply an overlay with the result, so no need to generate these cells.

Reply:
That makes sense. I understand that a "cell" should be filled with RGB values which are sent from mxcubecore. 

However the issue that we're seeing is that when we create a grid, the format of `this.gridResultFormat`  is `PNG`, not `RGB`, therefore the block below line 410 is not executed, and the results from a grid scan are not displayed. Probably there is a better solution to this, but in the current `develop` branch we can't display grid scan results

Reply:
Ah, hm yes its indeed a bit strange. I think some refactoring must have removed the bit that reads this from the backend. Its normally configurable in the beamline.yml file (https://github.com/mxcube/mxcubeweb/blob/develop/test/HardwareObjectsMockup.xml/beamline_config.yml#L81). So this should be set in the `sampleview` reducer and then this should set `setGridResultFormat` on the `DrawGridPlugin`.

It ought to be a quite easy fix, I wont have time to look at it right away. But feel free to give it a go if you feel inspired :). Id prefer that we fix the reading of this setting from the backend but feel free if you have some other ideas :) !?

Reply:
Thanks! changing the `mesh_result_format` to  `RGB` partially solved the issue. 

The only remaining issue I have is that after trying to save a grid which does not have results in it throws an error (see the image below). I updated the PR which basically checks for the length of the results before executing the code of line 325. 

Let me know if this is also something that can be fixed in another way, thank you!

![error_mxcube](https://github.com/mxcube/mxcubeweb/assets/49014169/d654014e-51f8-49f3-9331-549fd23aa5a8)

Reply:
Its difficult to say from this trace, maybe the output in the JS console will tell us more at what line the error occurred ?

Reply:
The error occurs [here](https://github.com/mxcube/mxcubeweb/blob/develop/ui/src/components/SampleView/DrawGridPlugin.js#L323) . 

The output I'm getting from the error is: 
```
ERROR
Cannot read properties of undefined (reading '1')
TypeError: Cannot read properties of undefined (reading '1')
    at DrawGridPlugin.cellFillingFromData (http://localhost:3000/static/js/bundle.js:25231:78)
    at DrawGridPlugin.shapeFromGridData (http://localhost:3000/static/js/bundle.js:25290:38)
    at http://localhost:3000/static/js/bundle.js:27971:52
    at Array.forEach (<anonymous>)
    at SampleImage.renderSampleView (http://localhost:3000/static/js/bundle.js:27964:26)
    at SampleImage.componentDidUpdate (http://localhost:3000/static/js/bundle.js:27357:10)
    at commitLifeCycles (http://localhost:3000/static/js/bundle.js:208274:28)
    at commitLayoutEffects (http://localhost:3000/static/js/bundle.js:210601:11)
    at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:193950:18)
    at Object.invokeGuardedCallbackDev (http://localhost:3000/static/js/bundle.js:193994:20)
```

It looks like the error occurs because `result` is an array of length 0 and we try to access `result[index][1]`. Adding another condition to the `if` statement ensuring that `Object.values(result).length > 0` seems to resolve the issue